PostProcessing/Runtime/PostProcessProfile.cs

```diff
@@ -1,131 +1,187 @@
-            effect.name = type.Name;
-            effect.enabled.value = true;
-            settings.Add(effect);
-            isDirty = true;
-            return effect;
-        }
-
-        public PostProcessEffectSettings AddSettings(PostProcessEffectSettings effect)
-        {
-            if (HasSettings(settings.GetType()))
-                throw new InvalidOperationException("Effect already exists in the stack");
-
-            settings.Add(effect);
-            isDirty = true;
-            return effect;
-        }
-
-        public void RemoveSettings<T>()
-            where T : PostProcessEffectSettings
-        {
-            RemoveSettings(typeof(T));
-        }
-
-        public void RemoveSettings(Type type)
-        {
-            int toRemove = -1;
-
-            for (int i = 0; i < settings.Count; i++)
-            {
-                if (settings[i].GetType() == type)
-                {
-                    toRemove = i;
-                    break;
-                }
-            }
-
-            if (toRemove < 0)
-                throw new InvalidOperationException("Effect doesn't exist in the stack");
-
-            settings.RemoveAt(toRemove);
-            isDirty = true;
-        }
-
-        public bool HasSettings<T>()
-            where T : PostProcessEffectSettings
-        {
-            return HasSettings(typeof(T));
-        }
-
-        public bool HasSettings(Type type)
-        {
-            foreach (var setting in settings)
-            {
-                if (setting.GetType() == type)
-                    return true;
-            }
-
-            return false;
-        }
-
-        public T GetSetting<T>() where T : PostProcessEffectSettings
-        {
-            foreach (var setting in settings)
-            {
-                if (setting is T)
-                    return setting as T;
-            }
-            return null;
-        }
-
-        public bool TryGetSettings<T>(out T outSetting)
-            where T : PostProcessEffectSettings
-        {
-            var type = typeof(T);
-            outSetting = null;
-
-            foreach (var setting in settings)
-            {
-                if (setting.GetType() == type)
-                {
-                    outSetting = (T)setting;
-                    return true;
-                }
-            }
-
-            return false;
-        }
-    }
-}
+using System;
+using System.Collections.Generic;
+
+namespace UnityEngine.Rendering.PostProcessing
+{
+    /// <summary>
+    /// An asset holding a set of post-processing settings to use with a <see cref="PostProcessVolume"/>.
+    /// </summary>
+    /// <seealso cref="PostProcessVolume"/>
+    public sealed class PostProcessProfile : ScriptableObject
+    {
+        /// <summary>
+        /// A list of all settings stored in this profile.
+        /// </summary>
+        [Tooltip("A list of all effects currently set in the stack.")]

```

Comment:
Should be:

> current set in the profile

The stack is the complete internal state of post-processing effects, a profile is just a set of effect settings that you add to a volume that will contribute to what goes in and out of the global stack.